platforms/software/ivy/src/main/java/org/gradle/api/publish/ivy/tasks/PublishToIvyRepository.java

```diff
@@ -87,10 +85,10 @@ public PublishToIvyRepository() {
     /**
      * The publication to be published.
      *
-     * @return The publication to be published
+     * Currently only instances of IvyPublication are supported.
```

Comment:
Since we cast in the setter, I think this belongs there instead of on the getter

Reply:
There are plenty of places on Gradle API where we return a default implementation from a getter without notifying the user. That’s the point of an interface. 

It’s on the setter that this distinction matters